### PR TITLE
Add support for Loading, Error and Empty Messages and Android Permissions

### PR DESCRIPTION
**Pull request by sfreeman422**
Adds:
- Loading message
- Error message
- Empty message
- Check for Android Platform, if so, ask for necessary permissions in camera roll, then accessCameraRoll

### CODE REVIEW

File: components/camera/CameraView.js [hunk 1]
```diff
@@ -17,7 +30,20 @@ export default class CameraView extends React.Component {
...
       const data = await this.camera.takePictureAsync(options);
-      console.log(data.uri);
+      if (Platform.OS === "android") {
+        const granted = await PermissionsAndroid.request(
+          PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE
+        );
+        if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+          await CameraRoll.saveToCameraRoll(data.uri);
+        } else {
+          // This should probably be a toast notification since it is useless rn.
```
Comment (comountainclimber) [inline components/camera/CameraView.js:40]: 👍

File: components/gallery/PhotoLibrary.js [hunk 2]
```diff
@@ -7,29 +7,46 @@ import {
...
   }
 
   accessCameraRoll = () => {
-    console.log(CameraRoll);
     return CameraRoll.getPhotos({ first: 20 })
       .then(photos =>
         Promise.resolve({
-          photos: photos.edges
+          photos: photos.edges,
+          message: photos.edges.length === 0 && "No pictures found!"
```
Comment (comountainclimber) [inline components/gallery/PhotoLibrary.js]:
```suggestion
          message: !photos.edges.length && "No pictures found!"
```

File: components/gallery/PhotoLibrary.js [hunk 3]
```diff
@@ -7,29 +7,46 @@ import {
...
-    photos: []
+    photos: [],
+    message: "Loading..."
   };
 
   // TODO: maybe all children should just do this by default HOC???
   async componentDidMount() {
     this.props.shouldRenderBackButton(true, this.props.history);
-    this.setState(await this.accessCameraRoll());
+    if (Platform.OS === "android") {
```
Comment (comountainclimber) [inline components/gallery/PhotoLibrary.js:27]: This is fine for now but both this and the other permission check would ideally be composable (HOC, hook, or just an exported function) for reusability and readability in this component 🤷‍♂️

**Comment by comountainclimber** [comment]
Overall looks good to me just have some minor comments 👍would be cool to see that permission stuff abstracted out one layer higher

**Review by comountainclimber — APPROVED** [review]

File: components/gallery/PhotoLibrary.js (hunk 3 above)
Comment (sfreeman422) [reply components/gallery/PhotoLibrary.js:27]: :100:

**Pull request merged**